### DISCUSSION TITLE
Patterns for npm watch?

### DISCUSSION THREAD

**Discussion by mpistrang** [BretFisher/dockercon19#21] [Q&A]
Thank you so much for this write up of your talk.  It has been immensely helpful!

Do you have any patterns or suggestions for using npm watch (or other techniques) for local development in Docker on typescript projects?  Anything you can share here that meshes well with the your multi-stage approach would be greatly appreciated.

Thanks again!

**Comment by BretFisher** [comment]
I'm not an expert in TypeScript + Docker + Local Dev + nodemon, but I tried to tackle that in a few videos of my [Docker for Node.js course.](https://www.bretfisher.com/docker-mastery-for-nodejs/)

It requires the right combo of:
VS Code launch configs
Compose config
nodemon.json for nodemon config
tsconfig.json

Example here, but last I tried, debugging wasn't setting break points. I did quickly try remote debugging in container but still had the same issue.

https://github.com/BretFisher/docker-mastery-for-nodejs/tree/main/typescript

**Comment by BretFisher** [comment]
Let me know if you figure out more details
